randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py

```diff
@@ -1,37 +1,37 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+import typing
 
 from PySide6 import QtWidgets
 
 from randovania.games.samus_returns.layout import MSRConfiguration
+from randovania.gui.generated.preset_msr_patches_ui import Ui_PresetMSRPatches
 from randovania.gui.preset_settings.preset_tab import PresetTab
 
-if TYPE_CHECKING:
+if typing.TYPE_CHECKING:
     from randovania.game_description.game_description import GameDescription
     from randovania.gui.lib.window_manager import WindowManager
     from randovania.interface_common.preset_editor import PresetEditor
     from randovania.layout.preset import Preset
 
+_FIELDS = [
+    "elevator_grapple_blocks",
+    "area3_interior_shortcut_no_grapple",
+    "nerf_power_bombs",
+    "nerf_super_missiles",
+]
 
-class PresetMSRPatches(PresetTab):
+
+class PresetMSRPatches(PresetTab, Ui_PresetMSRPatches):
     def __init__(self, editor: PresetEditor, game_description: GameDescription, window_manager: WindowManager):
         super().__init__(editor, game_description, window_manager)
+        self.setupUi(self)
 
-        self.root_widget = QtWidgets.QWidget(self)
-        self.root_layout = QtWidgets.QVBoxLayout(self.root_widget)
-
-        self.include_extra_pickups_check = QtWidgets.QCheckBox(self.root_widget)
-        self.include_extra_pickups_check.setEnabled(True)
-        self.include_extra_pickups_check.setText("Include Extra Pickups")
-        self.root_layout.addWidget(self.include_extra_pickups_check)
-
-        self.include_extra_pickups_label = QtWidgets.QLabel(self.root_widget)
-        self.include_extra_pickups_label.setWordWrap(True)
-        self.include_extra_pickups_label.setText("Include some optional pickups.")
-        self.root_layout.addWidget(self.include_extra_pickups_label)
+        self.setCentralWidget(self.centralWidget)
 
-        self.setCentralWidget(self.root_widget)
+        # Signals
+        for f in _FIELDS:
+            self._add_persist_option(getattr(self, f"{f}_check"), f)
```

Comment:
It's calling `_add_persist_option` but `_add_persist_option` does nothing.
It only defines an inner function without using it and there is no other code.